### PR TITLE
doc: add current status of tinygo based on issue #2979

### PR DESCRIPTION
**Pull request by leongross**
This PR records the status of the `tinygo` integration for `u-root`.
It will be updated regularly to reflect the latest changes to the project.

### CODE REVIEW

File: tools/tinygobb/README.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
+# Status of tinygo in u-root
+
+Currently, many of the busybox commands fail to build using tinygo.
+This document aims to cover the current process of enabling all subcommands to be built using tinygo.
+While enabling more and more commands the list of working commands in this document will be updated.
+
+## building
+Since currently the tinygo support is lacking, many commands got the `!tinygo` build tag.
+This prohibits building, so anyone recreating the build steps should be aware that this has to be removed for each command.
+The list below is the result of building each subcommand individually for x86_64 linux.
```
Comment (10000TB) [inline tools/tinygobb/README.md:10]: Is it limited to x86-64 ? we need them working on arm64 too

Comment (leongross) [reply tools/tinygobb/README.md:10]: It is probably not limited to x86_64, but the current tests were executed on x86_64.
Maybe a matrix would be better to display this information.

Comment (rminnich) [reply tools/tinygobb/README.md:10]: I have found that if it works on x86_64, it also works on risc-v and arm.

**Review by rminnich — COMMENTED** [review]
I think we need the matrix. 

Also, it would be nice to see the tinygo binary size, compared to the regular binary size.

**Comment by leongross** [comment]
> Also, it would be nice to see the tinygo binary size, compared to the regular binary size.
Yes that would be very helpful. Since this is quite some data we want to analyze, it would probably be a good idea to build some tooling around that to minimize the manual work

**Comment by rminnich** [comment]
I think the tinygo size comparison could be part of the CI tasks.

**Comment by 10000TB** [comment]
+1 for matrix.

Once there is a matrix, with some data as an example to allow future iterations, I think this will then be good to go

**Comment by leongross** [comment]
This idea is good, but I think we should keep the overview simple for now and extend it over time with a matrix build overview and a ci integration.

**Comment by leongross** [comment]
CI fails due `TLS handshake timeout` ...

**Comment by 10000TB** [comment]
> CI fails due `TLS handshake timeout` ...

likely an issue already at HEAD. I'd find it hard to believe this markdown change has anything to do with that

**Review by 10000TB — APPROVED** [review]

File: tools/tinygobb/README.md (hunk 1 above)
Comment (10000TB) [reply tools/tinygobb/README.md:10]: Added a tracker item to add compatibility matrix.  https://github.com/orgs/u-root/projects/2/views/1?pane=issue&itemId=66662510.

**Comment by 10000TB** [comment]
it looks like we all agree there need a matrix, but there is no objection to allow this go in first.

Added a tracker / FR to add a compatibility matrix: https://github.com/orgs/u-root/projects/2/views/1?pane=issue&itemId=66662510.

I'll make sure the FR is completed before closing the tab on tinygo.

thanks,

**Pull request merged**